aten/src/ATen/native/EmbeddingBag.cpp

```diff
@@ -74,26 +75,51 @@ void index_select_add<float>(const Tensor &select_indices,
                              const Tensor &add_indices,
                              const Tensor &src,
                              Tensor &output,
-                             const Tensor& offsets) {
+                             const Tensor& offsets,
+                             bool new_offsets) {
   int64_t ddim = src.size(1);
   auto src_data = src.data_ptr<float>();
   auto select_indices_data = select_indices.data_ptr<int64_t>();
   auto output_data = output.data_ptr<float>();
 
   if (isFastPathIndexSelect(src, output)) {
-    caffe2::EmbeddingLookupIdx(
-      /*block_size=*/ddim,
-      /*output_size=*/offsets.numel(),
-      /*index_size=*/select_indices.numel(),
-      /*data_size=*/src.size(0),
-      /*input=*/src_data,
-      /*indices=*/select_indices_data,
-      /*offsets=*/offsets.data_ptr<int64_t>(),
-      /*weights=*/nullptr,
-      /*scale_bias=*/nullptr,
-      /*normalize_by_lengths=*/false,
-      /*out=*/output_data
-    );
+    if (new_offsets) {
+      auto output_size = offsets.numel() - 1;
+      at::parallel_for(
```

Comment:
why do you parallelize in this branch but not in another?

Reply:
I think it's still doable just that it's not as plain.